### ISSUE TITLE
Duplicate `WebSocketException` class in starlette and websockets

### ISSUE THREAD

**Issue by Kludex** [Kludex/uvicorn#2081]
### Discussed in https://github.com/encode/uvicorn/discussions/2080

<div type='discussions-op-text'>

<sup>Originally posted by **ukalwa** December  7, 2022</sup>
Hello,

It looks like there is a websocket exception base class in both starlette and websockets. Should they be treated as separate. I ran into an issue where I am catching all websocket exceptions (including ConnectionClosedError) raised in a fastapi endpoint and using `startlette.exceptions.WebSocketException` didn't work. I had to use `websockets.exceptions.WebSocketException` to catch them.

Should I be using `websockets.exceptions` or `starlette.exceptions`? FastAPI exports starlette.exceptions but they don't seem to work in this case.

----
Here is a simple contrived example:

```python
from fastapi import FastAPI, WebSocket, WebSocketException, WebSocketDisconnect
import websockets.exceptions as ex
from traceback import print_exc
import asyncio

app = FastAPI()

@app.websocket("/ws/test")
async def test_ws_endpoint(websocket: WebSocket):
    await websocket.accept()
    try:
        while True:
             await websocket.send_json("test")
             await asyncio.sleep(10)
    except (WebSocketDisconnect, WebSocketException) as e:
        # Should get the trace here but it doesn't
        print_exc()
    except (ex.WebSocketException) as e:
        # Instead, the trace is printed here
        print_exc()
    finally:
        await websocket.close()
```

Here is the stack trace:
```
Traceback (most recent call last):
  File "/home/test/app.py", line 11, in test_ws_endpoint
    await websocket.send_json(None)
  File "/home/test/.venv/lib/python3.10/site-packages/starlette/websockets.py", line 173, in send_json
    await self.send({"type": "websocket.send", "text": text})
  File "/home/test/.venv/lib/python3.10/site-packages/starlette/websockets.py", line 85, in send
    await self._send(message)
  File "/home/test/.venv/lib/python3.10/site-packages/starlette/middleware/exceptions.py", line 65, in sender
    await send(message)
  File "/home/test/.venv/lib/python3.10/site-packages/uvicorn/protocols/websockets/websockets_impl.py", line 311, in asgi_send
    await self.send(data)  # type: ignore[arg-type]
  File "/home/test/.venv/lib/python3.10/site-packages/websockets/legacy/protocol.py", line 635, in send
    await self.ensure_open()
  File "/home/test/.venv/lib/python3.10/site-packages/websockets/legacy/protocol.py", line 944, in ensure_open
    raise self.connection_closed_exc()
websockets.exceptions.ConnectionClosedError: received 1005 (no status code [internal]); then sent 1005 (no status code [internal])
```

To reproduce, the client connects to the socket and then disconnects.</div>

> [!IMPORTANT]
> - We're using [Polar.sh](https://polar.sh/encode) so you can upvote and help fund this issue.
> - We receive the funding once the issue is completed & confirmed by you.
> - Thank you in advance for helping prioritize & fund our backlog.

<a href="https://polar.sh/encode/uvicorn/issues/2081">
<picture>
  <source media="(prefers-color-scheme: dark)" srcset="https://polar.sh/api/github/encode/uvicorn/issues/2081/pledge.svg?darkmode=1">
  <img alt="Fund with Polar" src="https://polar.sh/api/github/encode/uvicorn/issues/2081/pledge.svg">
</picture>
</a>

**Comment by Kludex** [comment]
Example that doesn't use any specific web framework:
```py
import asyncio

from websockets import exceptions as ex
from wsproto.utilities import LocalProtocolError


async def app(scope, receive, send):
    await send({"type": "websocket.accept"})
    try:
        while True:
            await send({"type": "websocket.send", "text": "potato"})
            await asyncio.sleep(1)
    except ex.WebSocketException:
        print("Server exception leaked to application.")
    except LocalProtocolError:
        print("Server exception leaked to application.")
```

**Comment by Kludex** [comment]
This is solved on Starlette and Uvicorn since 0.27.0.
